spec/models/doi_spec.rb

```diff
@@ -31,6 +31,85 @@
     end
   end
 
+  describe "after_commit" do
+    let(:doi) { create(:doi, aasm_state: "findable") }
+    let(:sqs_client) { instance_double(Aws::SQS::Client) }
+
+    before do
+      allow_any_instance_of(DataciteDoi).to receive(:send_message)
+      allow(Aws::SQS::Client).to receive(:new).and_return(sqs_client)
+      allow(IndexJob).to receive(:perform_later)
+      allow(DataciteDoi).to receive_message_chain(:__elasticsearch__, :index_document)
+    end
+
+    context "On Update event" do
+      it "sends import message if relevant attributes are modified" do
+        travel_to(Time.zone.local(2023, 12, 14, 10, 7, 40)) do
+          expect(doi).to receive(:send_import_message).with(doi.to_jsonapi)
+
+          doi.update(related_identifiers: [{ "relatedIdentifier" => "new_identifier", "relatedIdentifierType" => "DOI", "relationType" => "IsPartOf" }])
+        end
+      end
+
+      it "sends import message if creators are modified" do
+        travel_to(Time.zone.local(2023, 12, 14, 10, 7, 40)) do
+          expect(doi).to receive(:send_import_message).with(doi.to_jsonapi)
+
+          doi.update(creators: [{ "nameType" => "Personal", "name" => "New Creator" }])
+        end
+      end
+
+      it "sends import message if funding_references are modified" do
+        travel_to(Time.zone.local(2023, 12, 14, 10, 7, 40)) do
```

Comment:
Not a blocker:
I'd be interested in hearing more about this `travel_to` method.

Reply:
Sure, `mock record` value and `actual record` value were giving me same data in expect call, but `updated_at` was showing mili second difference in time, as they created at different time, to avoid that I used time travel.

My test case was failing intermittently, because of time issue,  

```
  -    "updated"=>2023-12-14 09:55:48.000000000 +0000},
  +    "updated"=>2023-12-14 09:55:49.000000000 +0000}
```

With the `travel_to` method, you can "time travel" within your tests or specific parts of your code. It's like telling your program, "Hey, for the next little while, pretend it's a different time."

Reply:
This is a great tool to add to our testing toolbelt.  Thanks